### DISCUSSION TITLE
react-native formatting is different - debug mode on/off

### DISCUSSION THREAD

**Discussion by jonasgroendahl** [js-fns/date-fns#2841] [Q&A] [answered]
There seems to be different formats for when debugging is on (development) and off (production) in react-native using the latest version of date-fns, any way to make them the same? i sense some inconsistencies will arise when shipping the app from development to production.

I'm getting the error:  Invalid time value.

As shown in the video below I'm trying out this format:  '2020-12-29 01:15:00' and calling isValid on it in both debug/no debug mode.

https://user-images.githubusercontent.com/23399953/103490562-0338bc00-4e1d-11eb-8185-c68919f836c8.mov

**Comment by jasondibenedetto** [comment]
Was there any update on this one?

**Comment by fturmel** [comment, marked as answer]
I think this has noting to do with `date-fns`. I could reproduce the behavior within the interactive code example in React Native's "get started" page just using a native Date object. Works on web, breaks on iOS.

https://user-images.githubusercontent.com/4009209/122968413-844b2780-d359-11eb-9070-8352a88422ab.mov

The problem is that you are using a string in the Date constructor, which can behave differently depending on the runtime.

Per MDN:
> Note: Parsing of date strings with the Date constructor (and Date.parse(), which works the same way) is strongly discouraged due to browser differences and inconsistencies.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date

Couple solutions:

- use a numerical timestamp in the Date constructor
- use one of the parse functions in `date-fns` to safely parse your string. `parseISO` should work well here. https://date-fns.org/docs/parseISO

Good luck!

**Comment by fturmel** [comment]
Some additional details on the React Native runtimes here: https://reactnative.dev/docs/javascript-environment#javascript-runtime

Essentially, it's JavaScriptCore (Safari's JS engine) that runs prod and V8 (Chrome) while debugging. I get the exact same error in Safari.

**Comment by fturmel** [comment]
Going to move this to a discussion so it's easier for people to find.
